docs/migration/enact/migrating-to-enact-5.md

```diff
@@ -0,0 +1,217 @@
+React 19 introduces Actions, new API `use`, `ref` as a prop, `Context` as a provider and some deprecated APIs.
+
+#### Actions
+Actions in React 19 are new features designed to handle asynchronous transitions automatically, such as submitting a form or fetching data.
+
+For example, you can use `useTransition` to handle the pending state for you:
+
+```js
+function Button() {
+  const [isPending, startTransition] = useTransition();
+
+  const handleClick = () => {
+    startTransition(async () => {
+      await new Promise(() => {
+        setTimeout(resolve, 3000);
+      });
+      console.log('Async job done');
+    });
+  };
+
+  return (
+    <button onClick={handleClick} disabled={isPending}>
+      {isPending ? 'Loading' : 'Click Me'}
+    </button>
+  );
+}
+```
+The async transition will immediately set the `isPending` state to `true`, make the async request(s), and then switch `isPending` to `false` after any transitions. This allows you to keep the current UI responsive and interactive while the data is changing.
+
+Please take a look at the new hooks for Actions, such as `useActionState` or `useOptimistic`, too.
+
+#### `use`
+The `use` API can be used in a component to read the value of a `Promise` or `context`
+When used with a `Promise`, React will suspend until the promise resolves. When `use` is used to read context, `use` can be called inside loops or conditional statements unlike `useContext`.
+
+```js
+const CounterContext = createContext(0);
+
+function MyCounter() {
+  const count = use(CounterContext);
+
+  return <div>Count: {count}</div>;
+}
+```
+
+#### `ref` as a prop
+In React 19, you can now access ref as a prop for function components:
+
+```js
+function MyComponent({ref}) {
+  return <button ref={ref}>Button</button>;
+}
+// Usage
+<MyComponent ref={ref} />;
+```
+
+Function components no longer need `forwardRef` to expose a DOM node to a parent component. `forwardRef` will be deprecated and removed in future versions.
+
+#### `Context` as a provider
+`<Context>` can now be rendered as a provider. Note that `<Context.Provider>` will be deprecated in future versions.
+
+```js
+const CounterContext = createContext(0);
+
+function App({children}) {
+  return <CounterContext value={5}>{children}</CounterContext>;
+}
+```
+
+#### Removed deprecated React APIs and React DOM APIs
+* **`propTypes` and `defaultProps` for functions**: These are removed. Using `propType` will be silently ignored. Use ES6 default parameters instead of `defaultProps`. Class components will continue to support `defaultProps`.
+* **`ReactDOM.findDOMNode`**: This API is removed. Replace it with DOM refs.
+* **`react-dom/test-utils`**: Import `act` from `react` instead of `react-dom/test-utils`.
+
+For more details, please refer to [React 19 Upgrade Guide](https://react.dev/blog/2024/04/25/react-19-upgrade-guide#breaking-changes) and [React v19](https://react.dev/blog/2024/12/05/react-19) from React Blog.
```

Comment:
The upgrade guide URL shouldn't be https://react.dev/blog/2024/04/25/react-19-upgrade-guide

Reply:
I intended to point out the breaking changes but it's ok to have the normal link.